### PR TITLE
updated agent to bot and changed code to match requirements

### PR DESCRIPTION
**Pull request by Tdrachev**
Updated code to match haseeb's suggestions

### CODE REVIEW

**Review by dikel — APPROVED** [review]
LGTM

File: forta-agents-linked/src/agent.js [hunk 1]
```diff
@@ -18,11 +18,17 @@ function provideHandleTransaction(timeHandler) {
   return async function handleTransaction(txEvent) {
     const findings = [];
     const filtered = txEvent.filterLog(eventABI, contractAddress);
+    let aboveThreshold = [];
     filtered.forEach((tx) => {
-      timeHandler.addToList(tx.args.agentId);
+      const { name } = tx;
+      if (name == "AgentUpdated") {
```
Comment (haseebrabbani) [inline forta-agents-linked/src/agent.js:24]: i think we'll need to check for creation only here (not updates). when a bot is updated, the assigned scan node does not change

File: forta-agents-linked/src/agent.js [hunk 2]
```diff
@@ -18,11 +18,17 @@ function provideHandleTransaction(timeHandler) {
...
+        timeHandler.addToListUpdated(tx.args.agentId);
+      } else if (name == "Link") {
+        timeHandler.addToListLinked(tx.args.agentId);
+        aboveThreshold = timeHandler.checkIfPassedThreshold();
```
Comment (haseebrabbani) [inline forta-agents-linked/src/agent.js:28]: we're only checking if the threshold is passed for Link events, but what if the Link event never fires? we would want to be alerted about this case i.e. agent is created, but does not get linked for longer than threshold

File: forta-agents-linked/README.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 ## Description
 
-Detects if a FORTA Bot has been assigned in a predefined threshold
+Detects if a FORTA Bot has been linked over 5 minutes after being updated
```
Comment (stoqnkpL) [inline forta-agents-linked/README.md:5]:
```suggestion
Detects if a FORTA Bot has not been linked over 5 minutes after being created
```

File: forta-agents-linked/package.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "forta-bot-assigned",
   "version": "0.0.1",
-  "description": "Detects if a FORTA Bot has been assigned",
+  "description": "Detects if a FORTA Bot has been linked over 5 minutes after being updated",
```
Comment (stoqnkpL) [inline forta-agents-linked/package.json:4]:
```suggestion
  "description": "Detects if a FORTA Bot has not been linked over 5 minutes after being created",
```

File: forta-agents-linked/src/TimeHandler.js [hunk 5]
```diff
@@ -8,21 +8,39 @@ Logic:
...
     this.threshold = threshold;
-    this.agentAddresses = {};
+    this.agentCreated = {};
+    this.agentLinked = {};
     this.findings = [];
   }
 
-  addToList(address) {
-    this.agentAddresses[address] = Math.floor(new Date().getTime() / 1000);
+  addToListUpdated(address) {
```
Comment (stoqnkpL) [inline forta-agents-linked/src/TimeHandler.js:16]: `addToListCreated` or `addToCreatedBotsList`

**Review by stoqnkpL — CHANGES_REQUESTED** [review]
Did we resolve this comment?
```
we're only checking if the threshold is passed for Link events, but what if the Link event never fires? we would want to be alerted about this case i.e. agent is created, but does not get linked for longer than threshold
```

**Review by stoqnkpL — APPROVED** [review]
LGTM!

**Pull request merged**